modules/core/src/test/java/org/apache/ignite/internal/processors/cache/GridCacheAbstractMetricsSelfTest.java

```diff
@@ -1403,4 +1409,72 @@ public void testInvokeAllAsyncMultipleKeysAvgTime() throws IgniteCheckedExceptio
 
         assertTrue(cache.localMetrics().getEntryProcessorAverageInvocationTime() > 0.0);
     }
+
+    /** */
+    @Test
+    public void testGetHistogram() {
+        IgniteCache<Integer, Integer> cache = grid(0).cache(DEFAULT_CACHE_NAME);
+
+        HistogramMetric getHistogram = metric("GetHistogram");
+
+        assertTrue(Arrays.stream(getHistogram.value()).allMatch(v -> v == 0));
+
+        cache.put(1, 1);
+
+        assertTrue(Arrays.stream(getHistogram.value()).allMatch(v -> v == 0));
+
+        cache.get(1);
+
+        assertTrue(Arrays.stream(getHistogram.value()).anyMatch(v -> v > 0));
+    }
+
+    /** */
+    @Test
+    public void testPutHistogram() {
+        IgniteCache<Integer, Integer> cache = grid(0).cache(DEFAULT_CACHE_NAME);
+
+        HistogramMetric putHistogram = metric("PutHistogram");
+
+        assertTrue(Arrays.stream(putHistogram.value()).allMatch(v -> v == 0));
+
+        cache.put(1, 1);
+
+        assertTrue(Arrays.stream(putHistogram.value()).anyMatch(v -> v > 0));
```

Comment:
Seems, here `v` should be exactly equal to one, itsn't it?

Reply:
Done.